### DISCUSSION TITLE
Potential Bug Found - User Authentication/Login

### DISCUSSION THREAD

**Discussion by connallykendrick** [Glench/ExtPay#63] [Q&A]
I think I've come to realize I'm really good at breaking things... explains my childhood 
and never being able to keep toys together. Lol.

I started testing with ExtPay a few days ago and just tonight experimented with the login screen in dev mode.

Preface - 
I don't know if any reference is needed here for the code I'm using, but it's pretty much the same
as what's laid out in the walkthrough.

Everything works great when it comes to (in dev mode...) -

- Install -> Payment/Dev-Login -> Ext updates to Active Usability
- Update user to "unpaid" status of whatever sort -> Ext updates to relevant disabling popup html
- Reactivate and update user status -> Ext updates to Active usability once again, as designed..
     **Successfully detecting whether or not the user has paid and is in active standing**

Here's where the problem lies...

- Install -> User Login -> Email Verification -> Success -> Ext updates to Active Usability
- Update user to "unpaid" status of whatever sort -> Ext does **not** update to relevant disabling popup html     
     **Now fails to detect whether or not the user has paid and is in active standing**

Result:
User can continue to use extension **even if** status is now unpaid/cancelled/past due 
when they've logged in initially as an existing, active user.

I further tested this by loading the extension in, on two different devices. 
One I activated as normal, while the other I "logged in."
Upon changing the user status **from** the "logged in" device, **normal** updated properly 
and **logged in** failed to do so. 

I'm in no understanding of how or why this is happening from the server side of things (or wherever), so I'll leave that up for further investigation... If we need anymore details, I'm here to provide.

Happy to be making progress
-Connally

**Comment by Glench** [comment]
This is probably only happening in development. If it happens in production please let me know.

**Reply by connallykendrick** [reply]
I later thought about this as well - I'll update you and this thread once tested outside of development
